execution_engine_testing/tests/benches/auction_bench.rs

```diff
@@ -0,0 +1,297 @@
+
+        let delegation_amount = U512::from(DELEGATION_AMOUNT);
+        let delegator_account_hash = delegator_public_key.to_account_hash();
+        let delegate = {
+            let contract_hash = builder.get_auction_contract_hash();
+            let entry_point = auction::METHOD_DELEGATE;
+            let next_validator_key = next_validator_iter
+                .next()
+                .expect("should produce values forever");
+            let args = runtime_args! {
+                auction::ARG_DELEGATOR => delegator_public_key.clone(),
+                auction::ARG_VALIDATOR => next_validator_key.clone(),
+                auction::ARG_AMOUNT => delegation_amount,
+            };
+            let mut rng = rand::thread_rng();
+            let deploy_hash = rng.gen();
+
+            let deploy = DeployItemBuilder::new()
+                .with_address(delegator_account_hash)
+                .with_stored_session_hash(contract_hash, entry_point, args)
+                .with_empty_payment_bytes(
+                    runtime_args! { ARG_AMOUNT => U512::from(DEFAULT_DELEGATE_COST), },
+                )
+                .with_authorization_keys(&[delegator_account_hash])
+                .with_deploy_hash(deploy_hash)
+                .build();
+
+            ExecuteRequestBuilder::new().push_deploy(deploy)
+        }
+        .build();
+        builder.exec(delegate);
+        builder.expect_success();
+        builder.commit();
+        builder.clear_results();
+    }
+
+    let mut era_end_timestamp = TIMESTAMP_INCREMENT_MILLIS;
+
+    // TODO: use add_bid to add non-genesis validator with stake
+
+    // advance the auction past the auction delay so that the added validator will be present in the
+    // auction
+    for _ in 0..DEFAULT_AUCTION_DELAY {
+        step_and_run_auction(&mut builder, &validator_keys);
+    }
+
+    group.bench_function(
+        format!(
+            "run_auction/validators/{}/delegators/{}",
+            validator_count, delegator_count
+        ),
+        |b| {
+            b.iter(|| {
+                era_end_timestamp += TIMESTAMP_INCREMENT_MILLIS;
+                step_and_run_auction(&mut builder, &validator_keys)
+            })
+        },
+    );
+}
+
+fn generate_pks(key_count: usize) -> Vec<PublicKey> {
+    let mut ret = Vec::with_capacity(key_count);
+    for _ in 0..key_count {
+        let bytes: [u8; SecretKey::ED25519_LENGTH] = rand::random();
+        let secret_key = SecretKey::ed25519_from_bytes(&bytes).unwrap();
+        let public_key = PublicKey::from(&secret_key);
+        ret.push(public_key);
+    }
+    ret
+}
+
+fn step_and_run_auction(builder: &mut LmdbWasmTestBuilder, validator_keys: &[PublicKey]) {
+    let mut step_request_builder = StepRequestBuilder::new()
+        .with_parent_state_hash(builder.get_post_state_hash())
+        .with_protocol_version(ProtocolVersion::V1_0_0)
+        .with_reward_item(RewardItem::new(DEFAULT_ACCOUNT_PUBLIC_KEY.clone(), 1));
```

Comment:
Is this still needed? We're iterating over all `validator_keys` and adding rewards to each. We can make all validators from `validator_keys` the _genesis validators_.

Reply:
You're right, we don't need to do this now that we have many validators.